### PR TITLE
refactor(behavior_path_planner): change start planner's variable name

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
### <samp>🤖 Generated by Copilot at 528ea68</samp>

Refactored the start planner module in the behavior path planner package to improve clarity and readability. Renamed some structs, members, functions, and variables to better describe the logic of pulling over from a parking spot. Adjusted the module's compatibility with other scene modules.

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp [hunk 1]
```diff
@@ -59,10 +59,9 @@ struct PullOutStatus
   PlannerType planner_type{PlannerType::NONE};
   PathWithLaneId backward_path{};
   lanelet::ConstLanelets pull_out_lanes{};
-  bool is_safe_static_objects{false};   // current path is safe against static objects
+  bool found_pull_over_path{false};     // current path is safe against static objects
```
Comment (kosuke55) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp:62]:
```suggestion
  bool found_pull_out_path{false};     // current path is safe against static objects
```

and oter parts too

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp:62]: oh...
Thanks!

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/start_planner/start_planner_module.hpp]: changed in https://github.com/autowarefoundation/autoware.universe/pull/5390/commits/3888e187a9408421827fe805900c915960414e5f

**Review by kosuke55 — APPROVED** [review]
thanks!!

**Pull request merged**